src/udf/python/risingwave/udf.py

```diff
@@ -49,14 +51,13 @@ class ScalarFunction(UserDefinedFunction):
     or multiple scalar values to a new scalar value.
     """
 
-    def __init__(self, *args, **kwargs):
-        self._io_threads = kwargs.pop("io_threads")
-        self._executor = (
-            ThreadPoolExecutor(max_workers=self._io_threads)
-            if self._io_threads is not None
-            else None
-        )
-        super().__init__(*args, **kwargs)
+    def __init__(self, io_threads=None, workers=None):
+        if io_threads is not None:
+            self._executor = ThreadPoolExecutor(max_workers=io_threads)
+        elif workers is not None:
+            self._executor = get_reusable_executor(max_workers=workers)
```

Comment:
I'm pondering whether `io_threads` can also be replaced by this...

Reply:
Yes. But using threads is more efficient than processes, because multiprocess needs pickling (serializing code) and data transfer between processes. That's why it's even slower than single thread mode for not very computationally heavy functions. 🤪